### PR TITLE
WIP: Introduce repository pattern

### LINKED ISSUE #380: Introduce Repository patten
https://www.amberbit.com/blog/2015/11/30/using-repository-pattern-to-migrate-rails-app-from-mongodb-to-postgresql/

### PR DESCRIPTION
**Pull request by dmitrytrager**
closes #380

### CODE REVIEW

**Comment by dmitrytrager** [comment]
@timurvafin should I submit PR into `examples` branch instead?

File: config/application.rb [hunk 1]
```diff
@@ -18,6 +18,7 @@ class Application < Rails::Application
 
     # The default locale is :en and all translations from config/locales/*.rb,yml are auto loaded.
     config.i18n.load_path += Dir[Rails.root.join("config", "locales", "**", "*.{rb,yml}")]
+    config.autoload_paths += Dir[Rails.root.join("app", "repositories", "*.rb")]
```
Comment (dmitrytrager) [inline config/application.rb]: @timurvafin do you think it's a good place to load files if they are not loaded by some gem?

**Comment by timurvafin** [comment]
@dmitrytrager need to discuss in person

**Comment by timurvafin** [comment]
Moved to https://github.com/fs/rails-base-examples/pull/1

**Pull request closed without merging**